src/lib/FreeC/Backend/Coq/Converter/Module.hs

```diff
@@ -65,12 +80,41 @@ convertImportDecl (IR.ImportDecl _ modName) = do
 --   Modules from the base library are imported via @From Base Require Import@
 --   sentences. Other external modules are imported via @From … Require@
 --   sentences, which means that references to these modules' contents must
---   be qualified in the code.
-generateImport :: Coq.ModuleIdent -> IR.ModName -> Converter Coq.Sentence
-generateImport libName modName = return
-  (mkRequireSentence libName [Coq.ident (showPretty modName)])
+--   be qualified in the code. For such an external module there is also an
+--   @Export … .QualifiedSmartConstructorModule@ sentence added if necessary,
+--   to give access to the notations for qualified smart constructors.
+generateImport :: Coq.ModuleIdent -> IR.ModName -> Converter [Coq.Sentence]
+generateImport libName modName = do
+  Just interface <- inEnv $ lookupAvailableModule modName
+  return (mkImportSentences [(Coq.ident (showPretty modName), interface)])
  where
-  -- | Makes a @From … Require Import …@ or  @From … Require …@.
-  mkRequireSentence :: Coq.ModuleIdent -> [Coq.ModuleIdent] -> Coq.Sentence
-  mkRequireSentence | libName == Coq.Base.baseLibName = Coq.requireImportFrom
-                    | otherwise = Coq.requireFrom
+  -- | Makes a @From … Require Import …@ or @From … Require …@ and an
+  --   @Export … .QualifiedSmartConstructorModule@ if this local module exists.
+  mkImportSentences :: [(Coq.ModuleIdent, ModuleInterface)] -> [Coq.Sentence]
```

Comment:
Is it actually necessary to have a list as the argument? As far as I can tell, the function gets only called once with a singleton list. This would also allow to remove the pair in favor of two arguments.